spring-cloud-consul-config/src/main/java/org/springframework/cloud/consul/config/ConsulPropertySource.java

```diff
@@ -53,25 +55,91 @@ public ConsulPropertySource(String context, ConsulClient source, String aclToken
 
 	public void init() {
 		Response<List<GetValue>> response;
-		if (aclToken == null) {
+		if (consulConfigProperties.getAclToken() == null) {
 			response = source.getKVValues(context, QueryParams.DEFAULT);
 		} else {
-			response = source.getKVValues(context, aclToken, QueryParams.DEFAULT);
-		}
-		List<GetValue> values = response.getValue();
-
-		if (values != null) {
-			for (GetValue getValue : values) {
-				String key = getValue.getKey();
-				if (!StringUtils.endsWithIgnoreCase(key, "/")) {
-					key = key.replace(context, "").replace('/', '.');
-					String value = getDecoded(getValue.getValue());
-					properties.put(key, value);
-				}
-			}
+			response = source.getKVValues(context,
+													consulConfigProperties.getAclToken(),
+													QueryParams.DEFAULT);
 		}
+
+		final List<GetValue> values = response.getValue();
+      final ConsulConfigFormat consulConfigFormat =
+         ConsulConfigFormat.fromString(consulConfigProperties.getConsulConfigFormat());
+      if (consulConfigFormat == ConsulConfigFormat.KEY_VALUE) {
+         parsePropertiesInKeyValueFormat(values);
+      } else if (consulConfigFormat == ConsulConfigFormat.PROPERTIES) {
+         parsePropertiesInPropertiesFormat(values);
+      }
 	}
 
+   /**
+    * Parses the properties in key value style i.e., values are expected to be either a sub key or a
+    * constant
+    *
+    * @param values
+    */
+   private void parsePropertiesInKeyValueFormat(List<GetValue> values) {
+      if (values == null) {
+         return;
+      }
+
+      for (GetValue getValue : values) {
+         String key = getValue.getKey();
+         if (!StringUtils.endsWithIgnoreCase(key, "/")) {
+            key = key.replace(context, "").replace('/', '.');
+            String value = getDecoded(getValue.getValue());
+            properties.put(key, value);
+         }
+      }
+   }
+
+   /**
+    * Parses the properties in key value style i.e., values are expected to be either a sub key or a
+    * constant
+    *
+    * @param values
+    */
+   private void parsePropertiesInPropertiesFormat(List<GetValue> values) {
+      if (values == null) {
+         return;
+      }
+
+      for (GetValue getValue : values) {
+         String key = getValue.getKey().replace(context, "");
+         if (!consulConfigProperties.getConsulConfigPropertiesKey().equals(key)) {
+            continue;
+         }
+         final String value = getDecoded(getValue.getValue());
+         // values should be key=value\nkey1=value....
+         final String[] propertyLines = value.split("\n");
```

Comment:
`StringToPropertiesConverter`, or just:
```
Properties props = new Properties();
// Must use the ISO-8859-1 encoding because Properties.load(stream) expects it.
props.load(new ByteArrayInputStream(source.getBytes("ISO-8859-1")));
```
directly.
The `TestPropertySourceUtils.convertInlinedPropertiesToMap()` does the trick as well.